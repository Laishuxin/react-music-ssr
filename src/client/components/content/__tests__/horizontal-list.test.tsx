import React from 'react'
import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import HorizontalList, { HorizontalListProps } from '../horizontal-list'

describe('horizontal-list', () => {
  const fn = jest.fn()
  const props: HorizontalListProps<{ name: string }> = {
    list: [{ name: '0' }, { name: '1' }, { name: '2' }],
    onChange: fn,
  }

  beforeEach(() => {
    fn.mockClear()
  })
  it('should render the correct horizontal list', () => {
    const { getByTestId } = screen
    render(<HorizontalList data-testid='test-horizontal-list' {...props} />)
    const element = getByTestId('test-horizontal-list')
    expect(element).toMatchSnapshot()
  })

  it('should perform the correct onChange event', () => {
    const { getByTestId, getByText } = render(
      <HorizontalList data-testid='test-horizontal-list-2' {...props} />,
    )
    const element = getByTestId('test-horizontal-list-2')
    expect(element).toBeInTheDocument()
    const items = element.querySelectorAll('.horizontal-list-item')
    expect(items.length).toBe(props.list.length)

    expect(items[0]).toHaveClass('active')
    fireEvent.click(items[0])
    expect(fn).not.toHaveBeenCalled()

    const targetElement = getByText('2')
    expect(targetElement).toBeInTheDocument()
    fireEvent.click(targetElement)
    waitFor(() => {
      expect(targetElement).toHaveClass('active')
      expect(fn).toHaveBeenCalledWith(2)
    })
  })
})
