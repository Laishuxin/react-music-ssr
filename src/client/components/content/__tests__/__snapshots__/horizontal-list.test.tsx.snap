// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`horizontal-list should render the correct horizontal list 1`] = `
<div
  class="scroll-wrapper"
  data-testid="test-horizontal-list"
>
  <div
    class="scroll-content"
    style="width: 0px;"
  >
    <div
      class="horizontal-list flex flex-row"
    >
      <span
        class="capsule rounded-lg p-1 text-current border-2 border-current border-solid horizontal-list-item flex-shrink-0 text-sm cursor-pointer active"
      >
        0
      </span>
      <span
        class="capsule rounded-lg p-1 text-current border-2 border-current border-solid horizontal-list-item flex-shrink-0 text-sm cursor-pointer border-transparent text-black"
      >
        1
      </span>
      <span
        class="capsule rounded-lg p-1 text-current border-2 border-current border-solid horizontal-list-item flex-shrink-0 text-sm cursor-pointer border-transparent text-black"
      >
        2
      </span>
    </div>
  </div>
</div>
`;
